hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/RecordListBuilder.java

```diff
@@ -395,6 +396,38 @@ public void revertChildrenOf(@NonNull final SingleTransactionRecordBuilderImpl r
         }
     }
 
+    /**
+     * Reverts or removes all child transactions after the given checkpoint.
+     * If there are no following records in the checkpoint, it means that the revert was executed on the user transaction.
+     */
+    public void revertChildrenFrom(@NonNull RecordListCheckPoint checkPoint) {
+        // If there are no following transactions than this means that the revert was executed on the user transaction
+        if (checkPoint.lastFollowingRecord() == null) {
+            revertChildrenOf(userTxnRecordBuilder);
+            return;
+        }
+
+        // We get to here when the revert was executed on a child transaction
+        // We need to revert all children that were added after the child transaction that was reverted
+        revertChildrenOf((SingleTransactionRecordBuilderImpl) checkPoint.lastFollowingRecord());
+
+        // We also need to revert all preceding transactions that were added after the first preceding transaction
+        if (checkPoint.firstPrecedingRecord() != null) {
+            var indexOf = precedingTxnRecordBuilders.indexOf(checkPoint.firstPrecedingRecord());
```

Comment:
As in your comment, don't we want to revert only preceding records added **after** `checkPoint.firstPrecedingRecord()`?

I.e., shouldn't we use `indexOf + 1` here?

Reply:
Yes, thats true. Great catch! 🙏🏻 
Just commited the changes :)